pkg/cloud/openstack/machine/actuator.go

```diff
@@ -121,21 +112,23 @@ func (oc *OpenstackClient) Create(ctx context.Context, cluster *clusterv1.Cluste
 	}
 
 	// get machine startup script
-	machineSetupConfig, err := oc.machineSetupWatcher.GetMachineSetupConfig()
-	if err != nil {
-		return err
-	}
-	role, ok := machine.ObjectMeta.Labels["set"]
-	if !ok {
-		klog.Errorf("Check machine role err, treat as \"node\" by default")
-		role = machinesetup.MachineRoleNode
-	}
-	startupScript, err := machineSetupConfig.GetSetupScript(role)
-	if err != nil {
-		return err
+	var ok bool
+	userData := []byte{}
+	if providerSpec.UserDataSecret != "" {
+		userDataSecret, err := kubeClient.CoreV1().Secrets(machine.Namespace).Get(providerSpec.UserDataSecret, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+
+		userData, ok = userDataSecret.Data[UserDataKey]
+		if !ok {
+			return fmt.Errorf("Machine's userdata secret %v did not contain key %v", providerSpec.UserDataSecret, UserDataKey)
+		}
 	}
+
```

Comment:
Would it make sense to error out here if no `UserDataSecret` is provided? I think rendering userdata does not make much sense without script data.

Reply:
I'd prefer not to error but I agree we shouldn't render anything if userData is empty. One may decide to just not pass any userData and it should be a valid use case. I'll add a check in the if below. thanks